actix-session/src/cookie.rs

```diff
@@ -308,10 +308,10 @@ where
         Session::set_session(state.into_iter(), &mut req);
 
         Box::new(self.service.call(req).map(move |mut res| {
-            if let Some(state) = Session::get_changes(&mut res) {
-                res.checked_expr(|res| inner.set_cookie(res, state))
-            } else {
-                res
+            match Session::get_changes(&mut res) {
+                (SessionStatus::Changed, Some(state)) =>
```

Comment:
i think we should update cookie on `change` and `renew`, and delete cookie on `purge`

Reply:
done